pkg/otlp/model/source/source_provider.go

```diff
@@ -0,0 +1,51 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package source
+
+import (
+	"context"
+	"fmt"
+)
+
+// Kind of source
+type Kind string
+
+const (
+	// InvalidKind is an invalid kind. It is the zero value of Kind.
+	InvalidKind Kind = ""
+	// HostnameKind is a host source.
+	HostnameKind Kind = "host"
+	// AWSECSFargateKind is a serverless source on AWS ECS Fargate.
+	AWSECSFargateKind Kind = "task_arn"
+)
+
+// Source represents a telemetry source.
```

Comment:
Is this what we'd call a resource in OpenTelemetry? If so, wouldn't that be better? I'm still a newbie here ⚠️  !

Reply:
Kinda, it's the Datadog-version of a resource. The goal is to have a single datatype that can represent hosts and serverless functions. Previously we had the `RunningTagFromAttributes` and `HostnameFromAttributes`, but since something can't be considered both a host and serverless, this simplifies the implementation a bit.